CATROID-1618: REPLACE OLD HELP WIKI URL WITH NEW URL

moe-threeface opened the pull request:
*Please enter a short description of your pull request and add a reference to the Jira ticket.*

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Branch-and-Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

moe-threeface committed:
CATROID-1618: Implement dynamic help url mapping

Replace wiki help urls(draft version with mapping aüüroach)

implemented skeleton for dynamic URL creation

* the URL is divided into 3 parts: base url, category url and brick url
* those 3 parts are selected dynamically according to mapping tables.
  if there exists help for a brick, it will have a corresponding entry
  in the mapping table.
* if there is no help for the brick but the category is know the user
  will be redirected to the category page (same if category is unknown
  - user will be redirected to base wiki)

added first tests for event-category

* added all (7) bricks for german and english
* remaining tests, that do not have a mapping are currently skipped.
  for those we could simply check if the URL links to the correct category
  (this would require additional category-dictionaries).

CATROID-1618 progress Add dynamic help url mapping for brick documentation

CATROID-1618 add dynemric help url, fixed getHelp function

CATROID-1618 Implement dynamic help url mapping

moe-threeface committed:
CATROID-1618: REPLACE OLD HELP WIKI URL WITH NEW URL

dorianpercic reviewed (commented):
@moe-threeface If possible, please always include a TL;DR in the PR description and add the JIRA link. Furthermore, please follow the JIRA-style commit guideline and do not capitalize the commit message, example of good commit message: `IDE-123 Add user login functionality`

dorianpercic commented:
Rebased PR.

dorianpercic commented on catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/CategoryListFragment.java:
```diff
@@ -595,18 +595,7 @@ public void onOptionsMenuClick(String tag) {
-				return Constants.CATROBAT_SENSORS_WIKI_URL + language;
-		}
-		return null;
+		return "https://catrobat.org/docs/brickdocumentation/";
```
Agree!

moe-threeface committed:
Replace wiki help urls(draft version with mapping aüüroach)

reichli committed:
implemented skeleton for dynamic URL creation

* the URL is divided into 3 parts: base url, category url and brick url
* those 3 parts are selected dynamically according to mapping tables.
  if there exists help for a brick, it will have a corresponding entry
  in the mapping table.
* if there is no help for the brick but the category is know the user
  will be redirected to the category page (same if category is unknown
  - user will be redirected to base wiki)

reichli committed:
added first tests for event-category

* added all (7) bricks for german and english
* remaining tests, that do not have a mapping are currently skipped.
  for those we could simply check if the URL links to the correct category
  (this would require additional category-dictionaries).

moe-threeface committed:
CATROID-1618 progress Add dynamic help url mapping for brick documentation
